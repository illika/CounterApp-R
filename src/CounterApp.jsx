import { useState } from 'react';
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    const handledAdd = () => {
        // 1 forma
        // setCounter(counter + 1);
        // 2 forma
        setCounter((c) => c + 1);
    }

    const handledSubtract = () => {
        setCounter(counter - 1);
    }

    const handledReset = () => {
        setCounter(value);
    }

    return (<>
        <h1>CounterApp</h1>
        <h2> {counter} </h2>

        <button onClick={handledAdd}>
            +1
        </button>
        <button aria-label='btn-reset' onClick={handledReset}>
            Reset
        </button>
        <button onClick={handledSubtract}>
            -1
        </button>
    </>);
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}