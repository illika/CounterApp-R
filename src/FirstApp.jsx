import PropTypes from "prop-types"

export const FirstApp = ({title, subTitle, name}) => {

    return (
        <>
            <div data-testid="test-title">{ title }</div>
            <p>{ subTitle }</p>
            <p>{ name } </p>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

FirstApp.defaultProps = {
    //title: "No hay titulo",
    subTitle: "No hay subTitle",
    name: "illika"
}