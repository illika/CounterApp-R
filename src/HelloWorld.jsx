export function App() {
    // document.createElement
    return (<h1>Hola mundo !!</h1>);
}