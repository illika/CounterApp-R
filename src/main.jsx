import React from 'react';
import ReactDOM from "react-dom/client";

import { App } from "./HelloWorld";
import { FirstApp } from "./FirstApp";

import './styles.css';

ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp title="Hola, Illika" subtitle={123} />
    </React.StrictMode>
);