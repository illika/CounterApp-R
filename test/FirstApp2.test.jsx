import { getAllByText, render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => {

    const title = 'Hola, Soy Illika';
    const subTitle = 'Soy un Subtitulo';

    test('Debe hacer match con snapshot', () => {

        const { container } = render(< FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar el mensaje Hola, Soy Illika', () => {
        render(< FirstApp title={title} />);
        // ve la pantalla interna del body
        //screen.debug();
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('Debe de mostrar el titulo en h1', () => {
        render(< FirstApp title={title} />);
        expect(screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    });

    test('Debe de mostrar el subtitulo', () => { 
        render(< FirstApp title={title} subTitle={subTitle} />);
        expect(screen.getAllByText(subTitle).length).toBe(1);

     })

});