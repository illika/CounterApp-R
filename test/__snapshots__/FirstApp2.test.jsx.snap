// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Illika
  </h1>
  <p>
    No hay subTitle
  </p>
  <p>
    illika
     
  </p>
</div>
`;
